pkg/operator/csi/csidrivercontrollerservicecontroller/helpers.go

```diff
@@ -63,3 +65,23 @@ func WithSecretHashAnnotationHook(
 		return nil
 	}
 }
+
+// WithReplicasHook sets the deployment.Spec.Replicas field according to the number
+// of available nodes. The number of nodes is determined by the node selector
+// specified in the field deployment.Spec.Templates.NodeSelector.
+func WithReplicasHook(nodeLister corev1listers.NodeLister) DeploymentHookFunc {
```

Comment:
What's the benefit of Deployment over DaemonSet if we run one per master node?

Reply:
> What's the benefit of Deployment over DaemonSet if we run one per master node?

deployments can be evicted and they are shut down on drain.  The shutdown is important enough to warrant using a deployment *for masters*.  If you're running on all workers as well, I would use a daemonset.

You may want to doc when to use one or the other.